packages/grafana-ui/src/types/datasource.ts

```diff
@@ -377,6 +377,12 @@ export interface DataStreamState {
    */
   delta?: DataFrame[];
 
+  /**
+   * Signal that the event is for delta of the data and that listener should handle
+   * any buffering.
+   */
+  isDelta?: boolean;
```

Comment:
Added this already here and in PanelData so I can check it and decide whether the ResultProcessor in explore needs to do any buffering with the data. At this moment I have a feeling that with the change in Loki streaming there isn't actually a DS that is sending delta data but not exactly sure, will need to check.

Reply:
Currently it is not delta events -- it is a buffered response.

If you want buffered results, we can add the key back to the request and response -- but I don't see why we need it right now

Reply:
We need in the response, so that ResultProcessor does not try to buffer it again. Otherwise I would need to say something like `if ds = loki: do not buffer` but hardcoding like that is not great and I am not sure about other datasources. If no datasource does deltas right now I can remove it and the whole delta handling branch in the ResultsProcessor. If there is a DS that does deltas I can just return a flag in the response. As I said I am not sure at the moment.

Reply:
> If no datasource does deltas right now

This is the only one

Reply:
and I think we can/should avoid it until we have a more clear API/structure

Reply:
Cool in that case I can simplify some of the code tomorrow.